tests/test_cnes_sonarqube.py

```diff
@@ -0,0 +1,251 @@
+                                  the image.
+        SONARQUBE_ADMIN_PASSWORD: the password of the admin account.
+        SONARQUBE_URL: URL of lequal/sonarqube container if already running
+                       without trailing / from the host.
+                       e.g. http://localhost:9000
+
+    The docstring of each test is the user story it tests.
+    """
+    # Class variables
+    RUN = os.environ.get('RUN', "yes")
+    SONARQUBE_CONTAINER_NAME = os.environ.get('SONARQUBE_CONTAINER_NAME', "lequalsonarqube")
+    SONARQUBE_ADMIN_PASSWORD = os.environ.get('SONARQUBE_ADMIN_PASSWORD', "adminpassword")
+    SONARQUBE_URL = os.environ.get('SONARQUBE_URL', "http://localhost:9000")
+
+    # Functions
+    @classmethod
+    def wait_cnes_sonarqube_ready(cls, container_name: str, tail = "all"):
+        """
+        This function waits for SonarQube to be configured by
+        the configure.bash script.
+
+        :param container_name: name of the container running lequal/sonarqube
+        :param tail: forwarded to docker logs
+        """
+        docker_client = docker.from_env()
+        while b'[INFO] CNES SonarQube: ready!' not in docker_client.containers.get(container_name).logs(tail=tail):
+            time.sleep(10)
+
+    @classmethod
+    def setup_class(cls):
+        """
+        Set up the tests
+        Launch a container and wait for it to be up
+        """
+        docker_client = docker.from_env()
+        # Launch a CNES SonarQube container
+        if cls.RUN == "yes":
+            print(f"Launching lequal/sonarqube container (name={cls.SONARQUBE_CONTAINER_NAME})...")
+            docker_client.containers.run("lequal/sonarqube:latest",
+                name=cls.SONARQUBE_CONTAINER_NAME,
+                detach=True,
+                auto_remove=True,
+                environment={"SONARQUBE_ADMIN_PASSWORD": cls.SONARQUBE_ADMIN_PASSWORD},
+                ports={9000: 9000})
+        else:
+            print(f"Using container {cls.SONARQUBE_CONTAINER_NAME}")
+        # Wait for the SonarQube server inside it to be set up
+        print(f"Waiting for {cls.SONARQUBE_CONTAINER_NAME} to be up...")
+        cls.wait_cnes_sonarqube_ready(cls.SONARQUBE_CONTAINER_NAME)
+
+    @classmethod
+    def teardown_class(cls):
+        """
+        Stop the container
+        """
+        if cls.RUN == "yes":
+            print(f"Stopping {cls.SONARQUBE_CONTAINER_NAME}...")
+            docker_client = docker.from_env()
+            docker_client.containers.get(cls.SONARQUBE_CONTAINER_NAME).stop()
+
+    def test_up(self):
+        """
+        As a user, I want the server to be UP so that I can use it.
+        """
+        status = requests.get(f"{self.SONARQUBE_URL}/api/system/status",
+                    auth=("admin", self.SONARQUBE_ADMIN_PASSWORD)).json()['status']
+        # Hint: if this test fails, the server might still be starting
+        assert status == "UP"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)